src/components/app/pageLocationKeyRaces/locationUnitedStatesLiveResults/keyRaceLiveResult.tsx

```diff
@@ -0,0 +1,348 @@
+    getIntlUrls(locale).locationUnitedStatesPresidential(),
+  )
+
+  const [candidateA, candidateB] = useInitialCandidateSelection(candidates)
+
+  const stateName = US_STATE_CODE_TO_DISPLAY_NAME_MAP[stateCode]
+  const raceName = primaryDistrict
+    ? `${stateName} ${formatDTSIDistrictId(primaryDistrict)} Congressional District Race`
+    : `${stateName} Senate Race`
+
+  const urls = getIntlUrls(locale)
+  const link = primaryDistrict
+    ? urls.locationDistrictSpecific({
+        stateCode: stateCode,
+        district: primaryDistrict,
+      })
+    : urls.locationStateSpecificSenateRace(stateCode)
+  const showLink = !isDistrictPage && !isSenatePage && !isPresidentialPage
+
+  const { data: liveResultData } = useApiDecisionDeskData({
+    initialRaceData,
+    stateCode,
+    district: primaryDistrict?.toString(),
+  })
+
+  const raceData = useMemo(() => {
+    if (!liveResultData) return null
+
+    if (primaryDistrict) {
+      return (
+        liveResultData?.find?.(race => {
+          return (
+            race.district.toString().toLowerCase() === primaryDistrict.toString().toLowerCase() &&
+            race.office?.officeId?.toString() === '3' &&
+            race.candidatesWithVotes.some(
+              _candidate =>
+                getPoliticianFindMatch(candidateA, _candidate) ||
+                getPoliticianFindMatch(candidateB, _candidate),
+            )
+          )
+        }) ?? null
+      )
+    }
+
+    return (
+      liveResultData?.find?.(race =>
+        race.candidatesWithVotes.some(
+          _candidate =>
+            getPoliticianFindMatch(candidateA, _candidate) ||
+            getPoliticianFindMatch(candidateB, _candidate),
+        ),
+      ) ?? null
+    )
+  }, [candidateA, candidateB, liveResultData, primaryDistrict])
+
+  const [ddhqCandidateA, ddhqCandidateB] = useLiveCandidateSelection(candidates, raceData)
+
+  const lastUpdated = useMemo(() => {
+    if (!raceData?.lastUpdated) return ''
+
+    const tzDate = new TZDate(raceData.lastUpdated, 'America/New_York')
+
+    return format(tzDate, "'Data updated' M/d/yy 'at' h:mm a 'ET'")
+  }, [raceData])
+
+  const raceStatus = useMemo<RaceStatus>(() => {
+    return getRaceStatus(raceData)
+  }, [raceData])
+
+  const winnerName = useMemo(() => {
+    if (!raceData) return ''
+    if (!raceData.calledCandidate) return ''
+    // @ts-expect-error
+    const firstName = raceData.calledCandidate?.first_name || raceData.calledCandidate?.firstName
+    // @ts-expect-error
+    const lastName = raceData.calledCandidate?.last_name || raceData.calledCandidate?.lastName
+    const fullName = `${firstName} ${lastName}`
+    if (fullName.includes('undefined')) return ''
+    return fullName
+  }, [raceData])
+
+  const canShowProgress = Boolean(liveResultData)
+
+  const isUncontested = useMemo(() => {
+    if (!raceData) return false
+    return raceData?.candidatesWithVotes.length === 1
+  }, [raceData])
+
+  if (isUncontested) {
+    return null
+  }
```

Comment:
This does not need to be memoized. This should do the trick.

```suggestion
  if (raceData?.candidatesWithVotes.length === 1) {
    return null
  }
```